src/states/elder/mod.rs

```diff
@@ -1574,6 +1571,12 @@ impl Approved for Elder {
             });
 
             self.send_neighbour_infos();
+
+            if elders_info.prefix() == &old_pfx {
```

Comment:
Probably better in 
```
} else if old_pfx.is_extension_of(elders_info.prefix()) {
} else if ...
```
like it is done in https://github.com/maidsafe/routing/pull/1798